s3_to_knack.py

```diff
@@ -220,13 +233,18 @@ def main():
 
     logging.info(f"{len(todos)} records to process.")
 
-    count = 1
-    for record in todos:
-        if count % 10 == 0:
-            logging.info(f"{count} record(s) processed")
-        method = "create" if not record.get("id") else "update"
-        app.record(data=record, method=method, obj=knack_obj)
-        count += 1
+    # Process todos with 10 workers
+    count = 0
+    with ThreadPoolExecutor(max_workers=10) as executor:
+        futures = []
+        for record in todos:
+            futures.append(
+                executor.submit(process_record, record, knack_obj, app, count)
+            )
+            count += 1
+
+    for future in futures:
+        future.result()  # This will re-raise any exceptions encountered in the threads
```

Comment:
nice!

[like i mentioned here](https://github.com/cityofaustin/atd-vz-data/pull/1478/files#r1668575664), akaik there's not a clean way to short-circuit this loop until all records are processed. this probably isn't a big deal 🤷